pkgs/applications/science/math/caffe/default.nix

```diff
@@ -0,0 +1,70 @@
+{ stdenv
+, openblas
+, boost
+, cudaSupport ? true
+, cudnnSupport ? false
+, cudnn ? null
+, cudatoolkit7
+, fetchzip
+, google-gflags
+, glog
+, hdf5
+, leveldb
+, lmdb
+, opencv
+, protobuf
+, snappy
+}:
+
+
+let optional = stdenv.lib.optional;
+in stdenv.mkDerivation rec {
+  # Use git revision because latest "release" is really old
+  name = "caffe-${rev}";
```

Comment:
Read this please https://nixos.org/nixpkgs/manual/#sec-package-naming and this https://github.com/NixOS/nixpkgs/pull/6539